app/components/StampSelector.tsx

```diff
@@ -152,21 +152,26 @@ export function StampSelector({ currentPlatform, currentProviders, verifiedProvi
 
 const useIncludedGroupsAndProviders = (specs: PlatformGroupSpec[]): Record<string, string[]> => {
   const customization = useCustomization();
+  const { stampScores } = useContext(ScorerContext);
 
   const included = useMemo(() => {
     const included: Record<string, string[]> = {};
-    specs.forEach((spec) => {
-      const providers = spec.providers?.map((p) => p.name) || [];
+    specs.forEach(({ platformGroup, providers }) => {
+      included[platformGroup] = providers.reduce((includedProviders, provider) => {
+        const hasCustomWeights = customization.scorer?.weights;
 
-      if (!customization.scorer?.weights) {
-        included[spec.platformGroup] = providers;
-        return;
-      }
+        // Hide if customization uses custom weights and this has a custom weight of 0
+        const checkHiddenByCustomization = () =>
+          hasCustomWeights && parseFloat(customization.scorer?.weights?.[provider.name] || "0") <= 0;
+
+        // Hide if deprecated and score is 0
+        const checkHiddenByDeprecation = () =>
+          provider.isDeprecated && parseFloat(stampScores[provider.name] || "0") <= 0;
 
```

Comment:
do `checkHiddenByCustomization` and `checkHiddenByDeprecation` need to be functions?
Why not compute the boolean value of the flag and store it in flags like `isHiddenByCustomization` and `isHiddenByDeprecation` ?

Reply:
Just to take advantage of short circuiting, especially if we add more conditions

Reply:
The question is what is cheaper?
a. to define a 2 functions on each iteration
b. or use if / else construct instead the short-circuit construct ...